tests/genmc/fail/data_race/atomic_ptr_write_dealloc_race.rs

```diff
@@ -0,0 +1,50 @@
+//@compile-flags: -Zmiri-genmc -Zmiri-disable-stacked-borrows
+
+// Test that use-after-free bugs involving atomic pointers are detected in GenMC mode.
+// This variant checks that the data race is still detected, even if the write happens before the free.
```

Comment:
It's not clear here what "this variant" is referring to -- what's the other variant?